src/modules/CurrencyModule.cpp

```diff
@@ -1,46 +1,689 @@
+				client.createMessage(message.channelId(), embed);
+				return;
+			}
+			if (args.at(1).toDouble() > gamblebetMax) 
+			{
+				client.createMessage(channel.id(), "You cannot bet an amount more than **" + QString::number(gamblebetMax) + config.currencySymbol+"**");
+				return;
+			}
+			selfGambleData.randNum = qrand() % (config.maxGuess - config.minGuess + 1) + config.minGuess;
+			selfGambleData.channelId = message.channelId();
+			selfGambleData.gamble = true;
+			selfGambleData.userId = message.author().id();
+			selfGambleData.doubleOrNothing = true;
+			selfGambleData.betAmount = args.at(1).toDouble();
+
+			Discord::Embed embed;
+			embed.setColor(qrand() % 16777216);
+			embed.setTitle("Welcome to Gamble (Double or Nothing)!");
+			embed.setDescription("All you need to do is guess a random number between " + QString::number(config.minGuess) + " and " + QString::number(config.maxGuess) + " (inclusive) and if it is the same as the number I guess, you get double the amount you just bet: **" + QString::number(2*selfGambleData.betAmount) + config.currencySymbol + "**!\n\n**What number do you think of?**");
+
+			client.createMessage(message.channelId(), embed);
+		}
+
+		//! Set a timer to reset if the player doesn't respond
+		auto timer = new QTimer();
+		timer->setSingleShot(true);
+		QObject::connect(timer, &QTimer::timeout, [this, &message, &client]() 
+			{
+				if (selfGambleData.gamble) {
+
+					selfGambleData.gamble = false;
+
+					client.createMessage(message.channelId(), "**Gamble Timeout due to no response.**");
+				}
+			});
+
+		timer->start(gambleTimeout * 1000);
+
+		});
+
+	RegisterCommand(Commands::CURRENCY_CLAIM, "claim", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel) 
+		{
+
+		QStringList args = message.content().split(' ');
+		GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+		QString prefix = setting->prefix;
+
+		if (args.first() != prefix + "claim")
+			return;
+
+		if (args.size() > 1) 
+		{
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+
+		if (args.size() == 1) 
+		{
+			if (!isRandomGiveAwayDone) 
+			{
+				if (allowGiveaway) 
+				{
+					Discord::Embed embed;
+					embed.setColor(qrand() % 11777216);
+					embed.setTitle("Claim FREEBIE");
+					embed.setDescription(":drum: And today's FREEBIE goes to **" + message.author().username() + "**! \n\n Congratulations! You just got **"+ QString::number(config.freebieReward) + config.currencySymbol +"**!");
+
+					m_settings[message.author().id()].currency += config.freebieReward;
+
+					client.createMessage(message.channelId(), embed);
+					isRandomGiveAwayDone = true;
+				}
+				else 
+				{
+					client.createMessage(message.channelId(), "**BRUH**, ***yOu CaN't JuSt GeT fReE sTuFf aNyTiMe.***");
+				}
+			}
+			else 
+			{
+				Discord::Embed embed;
+				embed.setColor(qrand()%11777216);
```

Comment:
As per the documentation, I would recommend using `RAND_MAX` here since it may very well differ between platforms.